skada/metrics.py

```diff
@@ -552,9 +552,21 @@ def _score(self, estimator, X, y, sample_domain=None):
         X, y, sample_domain = check_X_y_domain(X, y, sample_domain)
         source_idx = extract_source_indices(sample_domain)
 
-        backward_estimator = deepcopy(estimator)
-
-        y_pred_target = estimator.predict(X[~source_idx])
+        try:
+            backward_estimator = deepcopy(estimator)
+        except (TypeError, AttributeError):
+            try:
+                backward_estimator = clone(estimator)
+            except Exception:
+                raise ValueError(
```

Comment:
add a test here please that chack teh raise ?

Reply:
I would honestly remove this try/catch at all. The error message is somewhat misleading anyways.